sdk-tests/src/test/java/io/dapr/it/testcontainers/workflows/crossapp/WorkflowsCrossAppCallActivityIT.java

```diff
@@ -0,0 +1,267 @@
+  private static DaprContainer APP2_CONTAINER;
+  
+  // App3 container for App3FinalizeActivity - using DaprContainer with custom ports
+  private static DaprContainer APP3_CONTAINER;
+
+  // TestContainers for each app
+  private static GenericContainer<?> crossappWorker;
+  private static GenericContainer<?> app2Worker;
+  private static GenericContainer<?> app3Worker;
+
+  @BeforeAll
+  public static void setUp() throws Exception {
+    sharedPlacementContainer = new DaprPlacementContainer("daprio/placement:1.16.0-rc.3")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("placement")
+        .withReuse(false);
+    sharedPlacementContainer.start();
+    
+    sharedSchedulerContainer = new DaprSchedulerContainer("daprio/scheduler:1.16.0-rc.3")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("scheduler")
+        .withReuse(false);
+    sharedSchedulerContainer.start();
+
+    // Initialize Dapr containers with shared placement and scheduler
+    MAIN_WORKFLOW_CONTAINER = new DaprContainer(DAPR_RUNTIME_IMAGE_TAG)
+        .withAppName("crossapp-worker")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("main-workflow-sidecar")
+        .withPlacementContainer(sharedPlacementContainer)
+        .withSchedulerContainer(sharedSchedulerContainer)
+        .withComponent(new Component("kvstore", "state.in-memory", "v1", Map.of("actorStateStore", "true")))
+        .withComponent(new Component("pubsub", "pubsub.in-memory", "v1", Collections.emptyMap()))
+        .withSubscription(new Subscription("local", "pubsub", "topic", "/events"))
+        .withDaprLogLevel(DaprLogLevel.DEBUG)
+        .withLogConsumer(outputFrame -> System.out.println("MAIN_WORKFLOW: " + outputFrame.getUtf8String()))
+        .withAppChannelAddress("host.testcontainers.internal")
+        .waitingFor(Wait.forHttp("/v1.0/healthz/outbound")
+            .forPort(3500)
+            .forStatusCode(200)
+            .forStatusCode(204));
+    
+    APP2_CONTAINER = new DaprContainer(DAPR_RUNTIME_IMAGE_TAG)
+        .withAppName("app2")
+        .withNetwork(DAPR_NETWORK)
+        .withNetworkAliases("app2-sidecar")
+        .withPlacementContainer(sharedPlacementContainer)
+        .withSchedulerContainer(sharedSchedulerContainer)
+        .withCustomHttpPort(3501)
+        .withCustomGrpcPort(50002)
+        .withAppChannelAddress("main-workflow-sidecar:3500")
+        .withDaprLogLevel(DaprLogLevel.DEBUG)
+        .withComponent(new Component("kvstore", "state.in-memory", "v1",
+            Map.of("actorStateStore", "true")))
+        .withComponent(new Component("pubsub", "pubsub.in-memory", "v1", Collections.emptyMap()))
+        .withSubscription(new Subscription("local", "pubsub", "topic", "/events"))
+        .withLogConsumer(outputFrame -> System.out.println("APP2: " + outputFrame.getUtf8String()))
+        .waitingFor(Wait.forHttp("/v1.0/healthz/outbound")
```

Comment:
@cicoyle as stated before, this is already built in, there shouldn't be a need to define it here.